trantor/net/EventLoop.cc

```diff
@@ -170,44 +170,96 @@ void EventLoop::quit()
         wakeup();
     }
 }
+
+// The event loop needs a scope exit, so here's the simplest most limited
+// C++14 scope exit available (from
+// https://stackoverflow.com/a/42506763/3173540)
+//
+// TODO: If this is needed anywhere else, introduce a proper on_exit from, for
+// example, the GSL library
+namespace
+{
+template <typename F>
+struct ScopeExit
+{
+    ScopeExit(F &&f) : m_f(std::forward<F>(f))
+    {
+    }
+    ~ScopeExit()
+    {
+        m_f();
+    }
+    F m_f;
```

Comment:
Please follow the coding style of marking memory variables by append `_` at end of var name. Like `f_` or `functor_`